custom_components/tuya_local/devices/heatstorm_hs1500_heater.yaml

```diff
@@ -0,0 +1,132 @@
+name: Heatstorm heater
+products:
+  - id: gifk0oubmovuz3ky
+    name: HS-1500
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          value: heat
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 4
+        max: 37
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              value_redirect: set_temp_f
+              range:
+                min: 40
+                max: 99
+    - id: 3
+      type: integer
+      name: current_temperature
+      mapping:
+        - constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              value_redirect: cur_temp_f
+    - id: 4
+      type: string
+      name: preset_mode
+      mapping:
+        - dps_val: Auto
+          value: comfort
+        - dps_val: Low
+          value: eco
+        - dps_val: High
+          value: boost
+    - id: 13
+      type: bitfield
+      name: fault_code
+      mapping:
+        - dps_val: 1
+          value: Temp sensor fault
+        - dps_val: 2
+          value: Tilt over
+        - dps_val: 3
+          value: Over-voltage
+    - id: 19
+      type: string
+      name: temperature_unit
+      mapping:
+        - dps_val: c
+          value: C
+        - dps_val: f
+          value: F
+    - id: 20
+      type: integer
+      name: set_temp_f
+      range:
+        min: 40
+        max: 99
+      hidden: true
+    - id: 21
+      type: integer
+      name: cur_temp_f
+      hidden: true
+
+secondary_entities:
+  - entity: light
+    name: Display
+    category: config
+    dps:
+      - id: 5
+        name: brightness
+        type: string
+        mapping:
+          - dps_val: "Off"
+            value: 0
+          - dps_val: "10"
+            value: 26
+          - dps_val: "50"
+            value: 128
+          - dps_val: "100"
+            value: 255
+  - entity: lock
+    name: Child lock
+    icon: "mdi:hand-back-right-off"
+    category: config
+    dps:
+      - id: 7
+        type: boolean
+        name: lock
+        mapping:
+          - dps_val: true
+            value: false
+          - dps_val: false
+            value: true
+  - entity: binary_sensor
+    name: Fault
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 13
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+  - entity: select
+    name: Temperature unit
```

Comment:
name and icon should be replaced with `translation_key: temperature_unit` and values be in all lower case to use translations.